pkg/client/preflight.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+
+	version "github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/kubernetes"
+)
+
+var preflightChecks = []func(kubernetes.Interface) error{
+	preflightDNSCheck,
+	preflightVersionCheck,
+}
+
+// PreflightChecks runs all preflight checks in order, returning the first error encountered.
+func (c *SonobuoyClient) PreflightChecks() error {
+	client, err := c.Client()
+	if err != nil {
+		return err
+	}
+
+	for _, check := range preflightChecks {
+		if err := check(client); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+const (
+	kubeSystemNamespace = "kube-system"
+	kubeDNSLabelKey     = "k8s-app"
+	kubeDNSLabelValue   = "kube-dns"
+)
+
+func preflightDNSCheck(client kubernetes.Interface) error {
+	selector := metav1.AddLabelToSelector(&metav1.LabelSelector{}, kubeDNSLabelKey, kubeDNSLabelValue)
+
+	obj, err := client.CoreV1().Pods(kubeSystemNamespace).List(
+		metav1.ListOptions{LabelSelector: metav1.FormatLabelSelector(selector)},
+	)
+	if err != nil {
+		return errors.Wrap(err, "could not retrieve list of pods")
+	}
+
+	if len(obj.Items) == 0 {
+		return errors.New("no kube-dns tests found")
+	}
+
+	return nil
+}
+
+var (
+	minimumKubeVersion = version.Must(version.NewVersion("1.8.0"))
```

Comment:
I'd almost like to put all the version consts into the versions file, also I don' think we need a maximum.